paddle/framework/framework.proto

```diff
@@ -143,3 +143,13 @@ message BlockDesc {
 // https://github.com/PaddlePaddle/Paddle/blob/develop/doc/design/program.md
 // for more details.
 message ProgramDesc { repeated BlockDesc blocks = 1; }
+
+message OpPlacement {
+  optional string name = 1;
+  optional string device = 2;
```

Comment:
I am also wondering if device info for Operator is enough.
In Tensorflow,[ tf.Variable](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/state_ops.py#L96) is actually an operator, and [tf.Tensor](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/framework/ops.py#L306) has a operator data member. Tensorflow is a graph of operator, so device info in operator is enough. 
But we have both variable and operator.  Do we need device info for Variable? Do we need another `VarPlacement`?

Reply:
- In most common case, the output variable device is the same with operator device.
- For [control related operators](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/increment_op.cc) and [LoD related operators](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/lod_rank_table_op.cc), the operator device is always CPU. And the output variable is in CPU too.
- If we get training data/parameter using load operator, the variable device is the same with load operator.
- If we get training data/parameter using python reader, the variable device need to be set manually.

So, this a only one case which we should set device for variable. For other cases, the variable device can be decided by operator's device info.

Reply:
@QiJune thanks, great question, I guess we need `VarPlacement` only if we will use explicit OP for copying data from CPU to GPU?

Reply:
> If we get training data/parameter using python reader, the variable device need to be set manually.

Isn't the data initially CPU, and copied to GPU **implicitly** when needed, since we don't do explicit copies, maybe we don't need `VarPlacement`?